Framework/DataHandling/src/SavePDFGui.cpp

```diff
@@ -106,22 +117,28 @@ void SavePDFGui::exec() {
   // out << "#P0  -22.03808    1.10131 2556.26392    0.03422    1.50  0.5985\n";
   // // TODO
   out << "#L r G(r) dr dG(r)\n";
+}
 
-  // --------- write the data
+void SavePDFGui::writeWSData(std::ofstream &out,
+                             API::MatrixWorkspace_const_sptr inputWS) {
   const auto &x = inputWS->x(0);
   const auto &y = inputWS->y(0);
   const auto &dy = inputWS->e(0);
   HistogramData::HistogramDx dx(y.size(), 0.0);
   if (inputWS->sharedDx(0))
     dx = inputWS->dx(0);
   const size_t length = x.size();
-  for (size_t i = 0; i < length; ++i) {
-    out << "  " << x[i] << "  " << y[i] << "  " << dx[i] << "  " << dy[i]
-        << "\n";
+  if (x.size() == y.size()) {
```

Comment:
Could be simplified using the `Histogram` interface to the workspace and [points](https://docs.mantidproject.org/nightly/concepts/HistogramData.html#points)